### PR TITLE
Feat(inventory): Temp stash and custom drop custom time

### PR DESCRIPTION
**Pull request by FjamZoo**
This will close: https://github.com/overextended/ox_inventory/issues/1466

If you know a better method let me know 👍

### CODE REVIEW

**Comment by FjamZoo** [comment]
for use case something like

```lua
os.time() - 60 * 10  -- Stashes will now last 10 minutes extra
```

kinda shit implementation but no other good way I feel like

**Comment by thelindat** [comment]
- Would make more sense to pass the number of minutes you want it to last, rather than a timestamp.
- Still only a rough lifetime as it depends on the save frequency.
- Do people want it to timeout x time after it was last used, or after x time?

People could just make their own timeout/wait/loop/cronjob in their scripts and handle removal themselves using `RemoveInventory`.

**Comment by FjamZoo** [comment]
Yeah I feel like passing a time they want it to last would make more sense, would just require me to move inventoryClearTime a little up and then do calculations but it's still gonna be kinda ugly

Would look something like this
```lua
local function CustomDrop(prefix, items, coords, slots, maxWeight, instance, model, time)
	local dropId = generateInvId()
	local inventory = Inventory.Create(dropId, ('%s %s'):format(prefix, dropId:gsub('%D', '')), 'drop', slots or shared.playerslots, 0, maxWeight or shared.playerweight, false, {})

	if not inventory then return end

	inventory.items, inventory.weight = generateItems(inventory, 'drop', items)
	inventory.coords = coords

	if time then
		inventory.time = os.time() - (time * 60) - inventoryClearTime
	end

	Inventory.Drops[dropId] = {
		coords = inventory.coords,
		instance = instance,
		model = model,
	}

	TriggerClientEvent('ox_inventory:createDrop', -1, dropId, Inventory.Drops[dropId])

    return dropId
end
```

Or as you mentioned leave inventoryClearTime out and be a rough timeout instead of an absolote

**Comment by FjamZoo** [comment]
I made it a rough lifetime instead, easiest solution for now and worse case the inventory will just last a little longer

**Comment by thelindat** [comment]
> Do people want it to timeout x time after it was last used, or after x time?
>
> People could just make their own timeout/wait/loop/cronjob in their scripts and handle removal themselves using RemoveInventory (we just need an option to keep it alive).

Covers more use-cases. Could just have a negative value mean it doesn't get removed during save loop. Rather than just "time", something like "clearTime" to be more descriptive.

**Comment by FjamZoo** [comment]
> Covers more use-cases. Could just have a negative value mean it doesn't get removed during save loop.
If u want to switch to something like that I am all for it, just trying to get rid of some of these "suggestions".

I did do everything else u asked so lmk which route u want to go for.

File: modules/inventory/server.lua [hunk 1]
```diff
@@ -2660,6 +2665,10 @@ function Inventory.CreateTemporaryStash(properties)
 	inventory.items, inventory.weight = generateItems(inventory, 'drop', properties.items)
 	inventory.coords = coords
 
+	if properties.time then
```
Comment (thelindat) [inline modules/inventory/server.lua:2668]: `properties.clearTime`

**Comment by thelindat** [comment]
This doesn't even work right? inv.time gets updated everytime an inventory is accessed.

**Comment by mikigoalie** [comment]
> This doesn't even work right? inv.time gets updated everytime an inventory is accessed.

idk, but me likey the idea of setting time for custom stashes.. Perhaps time in minutes for it to disappear since last interaction and false / true / 0 for it to stay existing until restart or whatever? I'm working on a house robbery resource where I'm using temp stashes, if you want, I can rewrite this whole logic and make new pr

**Pull request closed without merging**